libfreerdp/codec/rfx.c

```diff
@@ -722,16 +781,15 @@ static BOOL rfx_process_message_tileset(RFX_CONTEXT* context, RFX_MESSAGE* messa
 	if (message->numTiles < 1)
 	{
 		WLog_ERR(TAG, "no tiles.");
-		return TRUE;
+		return FALSE;
 	}
 
 	Stream_Read_UINT32(s, tilesDataSize); /* tilesDataSize (4 bytes) */
 
-	context->quants = (UINT32 *)realloc((void*) context->quants, context->numQuant * 10 * sizeof(UINT32));
-	if (!context->quants)
+	if (!(pmem = realloc((void*) context->quants, context->numQuant * 10 * sizeof(UINT32))))
```

Comment:
@akallabeth I know but since it is in the context that would be unnecessary here. We return FALSE which in turn results in a NULL rfx message delivered to the outmost caller who might either continue to use the rfx context or free it.